File: defectmanager-web/src/service/PersonnelService.js
import PersonnelApi from '../network/api/PersonnelApi'

export default {
    getPersonnelList (projectId, success, error, failure) {
        PersonnelApi.getPersonnelList(projectId, success, error, failure)
    },
    addPersonnel (personnel, success, error, failure) {
        PersonnelApi.addPersonnel(personnel, success, error, failure)
    },
    updatePersonnel (personnel, success, error, failure) {
        PersonnelApi.updatePersonnel(personnel, success, error, failure)
    },
    removePersonnel (personnelId, success, error, failure) {
        PersonnelApi.removePersonnel(personnelId, success, error, failure)
    },
    getMyPersonnel (projectId, success, error) {
        let userId = localStorage.getItem('userId')
        PersonnelApi.getUserPersonnelOfProject(userId, projectId, success, error)
        // PersonnelApi.getPersonnelList(projectId, data => {
        //     for (let i = 0; i < data.length; i++) {
        //         if (data[i].userId == userId) {
        //             success(data[i])
        //             return
        //         }
        //     }
        // }, error)
    }
}

File: pesticide-server/src/main/java/pers/geolo/pesticide/service/PersonnelService.java
package pers.geolo.pesticide.service;

import pers.geolo.pesticide.entity.Personnel;

import java.util.List;

/**
 * @author 桀骜(Geolo)
 * @version 1.0
 * @date 2019/10/12
 */
public interface PersonnelService {

    Personnel getPersonnel(Integer personnelId);

    List<Personnel> getPersonnels(Integer projectId);

    void addPersonnel(Personnel personnel);

    void updatePersonnel(Personnel personnel);

    void removePersonnel(Integer personnelId);

    List<Personnel> getUserPersonnelOfProject(Integer userId, Integer projectId);
}
